apps/server/src/modules/task/uc/task.uc.spec.ts

```diff
@@ -1046,6 +1046,61 @@ describe('TaskUC', () => {
 		});
 	});
 
+	describe('revertPublished', () => {
+		let task: Task;
+
+		beforeEach(() => {
+			user = userFactory.buildWithId();
+			task = taskFactory.buildWithId({ creator: user });
+			userRepo.findById.mockResolvedValue(user);
+			taskRepo.findById.mockResolvedValue(task);
+			taskRepo.save.mockResolvedValue();
+		});
+
+		it('should check for permission to revert the task', async () => {
+			await service.revertPublished(user.id, task.id);
+			expect(authorizationService.checkPermission).toBeCalledWith(user, task, {
+				action: Actions.write,
+				requiredPermissions: [],
+			});
+		});
+
+		it('should throw ForbiddenException when not permitted', async () => {
+			const user2 = userFactory.buildWithId();
+			task = taskFactory.buildWithId({ creator: user2 });
+			taskRepo.findById.mockResolvedValue(task);
+			authorizationService.checkPermission.mockImplementation(() => {
+				throw new ForbiddenException();
+			});
+			await expect(async () => {
+				await service.revertPublished(user.id, task.id);
+			}).rejects.toThrow(ForbiddenException);
+		});
+
+		it('should revert the task back to draft state for the user', async () => {
+			task.unpublish = jest.fn();
+			await service.revertPublished(user.id, task.id);
+			expect(task.unpublish).toBeCalled();
```

Comment:
should we also check somehow more? Like, task.isPublished or isDraft or the prop private
I mean, the description should perhaps change to `should call unpublishe method in task entity`

Reply:
done 👍